### PR TITLE
HAProxy: update the recommended hardware configuration for HAProxy

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v8.0 (TiDB 8.0 versions)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.4 (TiDB 7.4 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/16463
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by djshow832 — APPROVED** [review]

File: best-practices/haproxy-best-practices.md [hunk 1]
```diff
@@ -35,12 +35,12 @@ Before you deploy HAProxy, make sure that you meet the hardware and software req
 
 ### Hardware requirements
 
-For your server, it is recommended to meet the following hardware requirements. You can also improve server specifications according to the load balancing environment.
+According to the [official documentation](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/), the minimum hardware configuration for HAProxy is shown in the following table. Under the sysbench `oltp_read_write` workload, the maximum QPS for this configuration is about 50K. You can also improve server specifications according to the load balancing environment.
```
Comment (qiancai) [inline best-practices/haproxy-best-practices.md:38]:
```suggestion
According to the [HAProxy documentation](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/), the minimum hardware configuration for HAProxy is shown in the following table. Under the Sysbench `oltp_read_write` workload, the maximum QPS for this configuration is about 50K. You can increase the server configuration according to your load balancing environment.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: best-practices/haproxy-best-practices.md [hunk 2]
```diff
@@ -35,12 +35,12 @@ Before you deploy HAProxy, make sure that you meet the hardware and software req
 
 ### Hardware requirements
 
-For your server, it is recommended to meet the following hardware requirements. You can also improve server specifications according to the load balancing environment.
+According to the [HAProxy documentation](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/), the minimum hardware configuration for HAProxy is shown in the following table. Under the sysbench `oltp_read_write` workload, the maximum QPS for this configuration is about 50K. You can also improve server specifications according to the load balancing environment.
```
Comment (hfxsd) [inline best-practices/haproxy-best-practices.md:38]:
```suggestion
According to the [HAProxy documentation](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/), the minimum hardware configuration for HAProxy is shown in the following table. Under the Sysbench `oltp_read_write` workload, the maximum QPS for this configuration is about 50K. You can increase the server configuration according to your load balancing environment.
```

Comment (qiancai) [inline best-practices/haproxy-best-practices.md:38]:
```suggestion
According to the [HAProxy documentation](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/), the minimum hardware configuration for HAProxy is shown in the following table. Under the Sysbench `oltp_read_write` workload, the maximum QPS for this configuration is about 50K. You can increase the server configuration according to your load balancing environment.
```

**Pull request merged**